### PR TITLE
FAQ: how to check whether variable is defined or not empty

### PR DESCRIPTION
**Pull request by GReagle**
Inspired and motivated by discussion on issue #2037.

### CODE REVIEW

File: sphinx_doc_src/faq.rst [hunk 1]
```diff
@@ -112,6 +112,28 @@ You can also declare a local variable in a block::
...
+How do I check whether a variable is defined?
+---------------------------------------------
+
+Use ``set -q var``.  For example, ``if set -q var; echo variable defined; end``.  To check multiple variables you can combine with ``and`` and ``or`` like so::
+
+    if set -q var1; or set -q var2
+        echo either variable defined
+    end
+
+Keep in mind that a defined variabled could be empty. Both of these commands define an empty variable: ``set var`` and ``set var ""``.
```
Comment (faho) [inline sphinx_doc_src/faq.rst]: These are still different, though!

```fish
$ set var
$ count $var
0
$ set var ''
$ count $var
1
```

But the underlying question there is, of course, how deep we want to go into it for a FAQ entry.

Do we want to go into `$var` vs `$var[1]`, do we want to add `set -ql`? Do we mention `test`?

See also [one of my more popular stackoverflow answers](https://stackoverflow.com/questions/47743015/fish-shell-how-to-check-if-a-variable-is-set-empty/47743269#47743269).

File: sphinx_doc_src/faq.rst [hunk 2]
```diff
@@ -112,6 +112,28 @@ You can also declare a local variable in a block::
...
+
+
+How do I check whether a variable is not empty?
```
Comment (faho) [inline sphinx_doc_src/faq.rst]: I see how you're trying to ignore `test`, but I think doing so in this context is a disservice to the readers.

It should at least be mentioned, in something like "for compatibility with other shells, there is also `test -n "$var"`.

**Comment by floam** [comment]
`test -n $arg` checks if the string is empty and is a lot less to remember, and without the user needing to wrap their head around `--` to stop option parsing. I think that should be taught by the FAQ entry.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**